src/point_one/fusion_engine/common/portability.h

```diff
@@ -45,6 +64,31 @@ typedef int64_t p1_ssize_t;
 typedef ssize_t p1_ssize_t;
 #endif
 
+#ifndef P1_HAVE_STD_OSTREAM
+#  define P1_HAVE_STD_OSTREAM 1
+#endif
+#if P1_HAVE_STD_OSTREAM
+#  include <ostream>
+using p1_ostream = std::ostream;
+#else
+class p1_ostream {
+ public:
+  p1_ostream() = default;
+};
+template <class T>
+inline p1_ostream& operator<<(p1_ostream& stream, const T&) {
```

Comment:
Does `<ostream>` define a default templated no-op operator like this? I would think you wouldn't want to have one, and it would be better to get a compiler error if you try to stream something where << isn't defined?

Reply:
Oh I think nevermind - this is effectively a cheap way to define no-ops for all the primitive `operator<<()` overloads that are defined in `basic_stream` normally.